dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java

```diff
@@ -272,45 +303,128 @@ public ContentletMetadata generateContentletMetadata(final Contentlet contentlet
      */
     @Override
-    public Map<String, Serializable> getMetadata(final Contentlet contentlet,final  String fieldVariableName)
+    public Metadata getMetadata(final Contentlet contentlet,final  String fieldVariableName)
             throws DotDataException {
 
-        final StorageType storageType = StoragePersistenceProvider.getStorageType();
-        final String metadataBucketName = Config.getStringProperty(METADATA_GROUP_NAME, DOT_METADATA);
-        final String metadataPath       = this.getFileName(contentlet, fieldVariableName);
+        return getMetadata(contentlet, fieldVariableName, false);
+    }
 
-        return this.fileStorageAPI.retrieveMetaData(
-                new RequestMetadata.Builder()
-                        .wrapMetadataMapForCache(this::filterNonCacheableMetadataFields)
-                        .cache(()-> contentlet.getInode() + StringPool.COLON + fieldVariableName)
-                        .storageKey(new StorageKey.Builder().group(metadataBucketName).path(metadataPath).storage(storageType).build())
-                        .build()
-        );
+    /**
+     * {@inheritDoc}
+     * @param contentlet          {@link Contentlet}
+     * @param fieldVariableName  {@link String}
+     * @return
+     */
+    @Override
+    public Metadata getMetadataForceGenerate(final Contentlet contentlet, final String fieldVariableName)
+            throws DotDataException {
+        return getMetadata(contentlet, fieldVariableName, true);
     }
 
+    /**
+     * {@inheritDoc}
+     * @param contentlet          {@link Contentlet}
+     * @param fieldVariableName  {@link String}
+     * @param forceGenerate  @boolean
+     * @return
+     */
+    private Metadata getMetadata(final Contentlet contentlet, final String fieldVariableName, final boolean forceGenerate)
+            throws DotDataException {
+
+        if(null != contentlet.get(fieldVariableName)) {
+            final StorageType storageType = StoragePersistenceProvider.getStorageType();
+            final String metadataBucketName = Config
+                    .getStringProperty(METADATA_GROUP_NAME, DOT_METADATA);
+            final String metadataPath = getFileName(contentlet, fieldVariableName);
+
+            Map<String, Serializable> metadataMap = fileStorageAPI.retrieveMetaData(
+                    new FetchMetadataParams.Builder()
+                            .projectionMapForCache(this::filterNonCacheableMetadataFields)
+                            .cache(() -> contentlet.getInode() + StringPool.COLON
+                                    + fieldVariableName)
+                            .storageKey(new StorageKey.Builder().group(metadataBucketName)
+                                    .path(metadataPath).storage(storageType).build())
+                            .build()
+            );
+
+            if (null != metadataMap) {
+                return new Metadata(fieldVariableName, metadataMap);
+            }
+
+            if (forceGenerate) {
```

Comment:
if we want to force the metadata generation, this if shouldn't be above? Otherwise, if metadataMap != null, this code won't be reached

Reply:
yeah, it's fine. if metadata == null 
it means it doesn't exist. Therefore we have the option to force its generation